scalanet/ut/src/io/iohk/scalanet/peergroup/DynamicTLSPeerGroupSpec.scala

```diff
@@ -349,6 +375,142 @@ class DynamicTLSPeerGroupSpec extends AsyncFlatSpec with BeforeAndAfterAll {
+      // we are using pretty large messages to be sure incoming and sending buffers will be full
+      val m1 = SizeAbleMessage.genRandomOfsizeN(16 * 1024 * 1024)
+      val m2 = SizeAbleMessage.genRandomOfsizeN(16 * 1024 * 1024)
+      for {
+        promise <- Deferred.tryable[Task, Option[Throwable]]
+        _ <- clientChannel.sendMessage(m1)
+        _ <- Task.sleep(1.second)
+        // snd buffer and rcv buffer are full, flushing message to snd buffer will be impossible
+        _ <- clientChannel.sendMessage(m2).doOnFinish(rest => promise.complete(rest)).startAndForget
+        _ <- Task.sleep(1.second)
+        promiseFinished <- promise.tryGet
+        _ <- Task(assert(promiseFinished.isEmpty))
+        exception <- Task.parMap2(srvChannelImpl.close(), promise.get)((_, ex) => ex)
+      } yield {
+        assert(exception.get.isInstanceOf[ChannelBrokenException[_]])
+      }
+  }
+
+  it should "successfully flush buffer when it will be free on receiver side " in backPressureTestCase(
+    clientConfig = getCorrectConfig(maxFrameLength = 64 * 1024 * 1024),
+    serverConfig = getCorrectConfig(maxQueueSize = 1, maxFrameLength = 64 * 1024 * 1024)
+  ) {
+    case (clientChannel, serverChannel) =>
+      // we are using pretty large messages to be sure incoming and sending buffers will be full
+      val m1 = SizeAbleMessage.genRandomOfsizeN(16 * 1024 * 1024)
+      val m2 = SizeAbleMessage.genRandomOfsizeN(16 * 1024 * 1024)
+      for {
+        finished <- Deferred.tryable[Task, Unit]
+        _ <- clientChannel.sendMessage(m1)
+        _ <- Task.sleep(1.second)
+        // snd buffer and rcv buffer all full, flushing message will be impossible
+        _ <- clientChannel.sendMessage(m2).doOnFinish(_ => finished.complete(())).startAndForget
+        _ <- Task.sleep(2.second)
+        check <- finished.tryGet
+        // even after some time message is not sent as buffers are still full
+        _ <- Task(assert(check.isEmpty))
+        // sending message should finish once receiver will consume event on its side and make space and rcv buffer
+        serverEvent1 <- Task.parMap2(finished.get, serverChannel.nextChannelEvent)((_, sEv) => sEv)
+        serverEvent2 <- serverChannel.nextChannelEvent
+      } yield {
+        assert(serverEvent1.get == MessageReceived(m1))
+        assert(serverEvent2.get == MessageReceived(m2))
+      }
+  }
+
+  it should "get more bytes form rcv buffer as soon as incoming queue will be at least half empty" in backPressureTestCase(
+    clientConfig = getCorrectConfig(maxFrameLength = 64 * 1024 * 1024),
+    serverConfig = getCorrectConfig(maxQueueSize = 4, maxFrameLength = 64 * 1024 * 1024)
+  ) {
+    case (clientChannel, serverChannel) =>
+      val srvChannelImpl =
+        serverChannel.asInstanceOf[DynamicTLSPeerGroupInternals.DynamicTlsChannel[ChannelEvent[SizeAbleMessage]]]
+      // we are using small messages to not fill up snd and rcv buffers
```

Comment:
Can these be configured explicitly?

Reply:
It is as I have explained in other comment